### DISCUSSION TITLE
Convert Jpeg to Avif

### DISCUSSION THREAD

**Discussion by Sammy190** [dlemstra/Magick.NET#1261] [Help]
Hello,

I am trying to convert a Jpeg image to Avif format using the latest version of Magick.NET

I tried to apply the following code : 

`magickImage.Quality = 80;`
`magickImage.Settings.SetDefine(MagickFormat.Avif, "method", "4");`
`magickImage.Format = MagickFormat.Avif;`

Which converts the image to Avif format, but with a terrible quality. And changing the Quality parameter or the method does not seem to have any effect (the size of the image file does not change).

I got the commands from this page : https://imagemagick.org/script/webp.php 
I know these commands are used for the Webp format and can be incompatible in my context, but I could not find anything about Avif in [the documentation ](https://imagemagick.org/script/formats.php)

**My question is :** 
Are there specific commands in Magick.NET to adjust the compression quality and speed on the Avif format, the same way we can do with the Webp format ?

**Comment by dlemstra** [comment]
You cannot use the WebP options and expect them to work. Setting the `Quality` property should change the quality of the output image. Can you share your input image that can be used to reproduce this?

**Reply by Sammy190** [reply]
Thank you for your quick answer.
 
So the quality parameter is generic to all the image formats. 
Here are the image I used + the outputs for a quality = 100 and quality = 20
[images.zip](https://github.com/dlemstra/Magick.NET/files/9646241/images.zip) (_I put a zip because I could not upload *.avif files_)

The resulting image is too blurry, especially around the red circle parts. I have far better results when using Webp. Is this normal ?   
Are not there other parameters like the _quality_ command to fine tune the Avif output

**Reply by dlemstra** [reply]
I am getting a different output image when I set the quality to 100 so I am not sure if we are doing the same thing. But it does look like `avif` images are not the best format for these type of images. There are no other options available right now for heic that have impact on the quality.

**Reply by Sammy190** [reply]
Could I ask what do you mean by "`avif` are not the best format for _these types of images_" ? Do you mean Avif is not suited for xray images ? 

I had another question if I may ask, I saw that the library had a method :
`ImageOptimizer().LosslessCompress(Stream)`
Which can be used for lossless compression but only for the Jpg, Png, Gif and Icon formats.

For the Webp format we have the command : `magickImage.Settings.SetDefine(MagickFormat.Webp, "lossless", "true")` that does the same I guess if we follow the description.

Do we have a similar command that I missed to apply a lossless compression on Avif ? Or is `quality = 100` applying implicitly a lossless compression ?

**Reply by Sammy190** [reply]
Hello, 
Sorry about the insistance but it's been two weeks and I didn't have an answer. 
Can you just give me more details about the first question please : "`avif` are not the best format for these types of images" ? What did you mean by that ? 
Thank you
